types/react/index.d.ts

```diff
@@ -2126,6 +2126,30 @@ declare namespace React {
      */
     export function startTransition(scope: TransitionFunction): void;
 
+    /**
+     * Wrap any code rendering and triggering updates to your components into `act()` calls.
+     *
+     * Ensures that the behavior in your tests matches what happens in the browser
+     * more closely by executing pending `useEffect`s before returning. This also
+     * reduces the amount of re-renders done.
+     *
+     * @param callback A synchronous, void callback that will execute as a single, complete React commit.
+     *
+     * @see https://reactjs.org/blog/2019/02/06/react-v16.8.0.html#testing-hooks
+     */
+    // While act does always return Thenable, if a void function is passed, we pretend the return value is also void to not trigger dangling Promise lint rules.
+    export function act(callback: () => VoidOrUndefinedOnly): void;
+    export function act<T>(callback: () => T | Promise<T>): Promise<T>;
+
+    /**
+     * @deprecated Use `act` instead (available from React 18.3.1 and later)
+     */
+    export function unstable_act(callback: () => VoidOrUndefinedOnly): void;
```

Comment:
Let's not clog up autocomplete with an unstable API for just one patch release cycle. It's especially problematic since these types would apply to `18.3.0` where this function wouldn't be available.

Reply:
Actually, `unstable_act` has been available since 18.0.0, and is the _only_ available method until 18.3.1.
In 18.3.1, both are available. Do you still want me to only include `act`?

Reply:
Thought about this some more. Given that React 18 has been out for two+ years and no one has complained about the missing `unstable_act`, I support not including it in the types 👍